### PR TITLE
Add drf

### PR DESCRIPTION
**Pull request by jnation3406**
Parts of this PR:

- Switch to using DRF
- Rename observatory->enclosure and updated->modified in the model
- Connect to obs portal for oauth authentication of posting data
- Connect to configdb for validation of site/enclosure/telescope
- Move api to /api and add frontend for viewing at /

The oauth authentication is currently just HTTPBasicAuth here, but I have a branch that supports getting and using a bearer token for auth. We are planning to revisit the auth model for all the OCS applications in a later sprint so that stuff may change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,12 +26,19 @@ This project is configured using environment variables.
...
+| `DEBUG`              | Django Debug mode                                                                  | False                        |
 | `DB_ENGINE`          | Database Engine, set to `django.db.backends.postgresql_psycopg2` to use PostgreSQL | `django.db.backends.sqlite3` |
 | `DB_NAME`            | Database Name                                                                      | `db.sqlite3`                 |
 | `DB_HOST`            | Database Hostname, set this when using PostgreSQL                                  | _empty string_               |
 | `DB_USER`            | Database Username, set this when using PostgreSQL                                  | _empty string_               |
 | `DB_PASS`            | Database Password, set this when using PostgreSQL                                  | _empty string_               |
 | `DB_PORT`            | Database Port, set this when using PostgreSQL                                      | `5432`                       |
+| `OAUTH_CLIENT_ID`    | OAuth authentication client id (found in observation portal admin for the app)     | ``                           |
```
Comment (timbeccue) [inline README.md:36]: Not sure who the target audience is for the readme, but it's not clear to me where I would find this value. Is it defined when creating the oauth server somewhere else? If it's something a new user trying to set things up (me) should be able to figure out, then additional details might be helpful.

**Review by timbeccue — APPROVED** [review]
Looks fine to me.

File: README.md (hunk 1 above)
Comment (jnation3406) [reply README.md:36]: The oauth client id and secret and autogenerated for you when you create an oauth "application" on the server. We should probably describe this better in overall project docs and reference from here...

Comment (eheinrich) [reply README.md:36]: Agreed, when we work out how we want to do oauth across the projects, I think that would be a good time to create some documentation on it.

**Review by eheinrich — COMMENTED** [review]
The tests fail for me, and I get a 500 when I navigate to `/api/`, until I run `python manage.py collectstatic`. The index page works however. Is there a way to serve static files for local development without having to run collectstatic first?

File: schedule/tests.py [hunk 2]
```diff
@@ -44,4 +45,62 @@ def test_downtime_with_end_before_start_not_allowed(self):
         self.assertEqual(Downtime.objects.count(), 0)
         response = self.client.post(reverse('admin:schedule_downtime_add'), data, follow=True)
         self.assertEqual(Downtime.objects.count(), 0)
-        self.assertIn('Start time must come before end time', str(response.content))
+        self.assertIn('End time must be after start time', str(response.content))
+
+
+class TestDowntimeSerializer(TestCase):
```
[the remark this answers is not in the document]
Comment (jnation3406) [reply schedule/tests.py:51]: good idea, I'll add it

File: schedule/templates/downtime_list.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
+    <title>Downtime Database</title>
+    <meta name="description" content="">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    {% bootstrap_css %}
+  </head>
+  <body>
+    <nav class="navbar navbar-expand-lg navbar-light navbar-bg head">
+      <div class="container ">
+        <span class=navbar-brand><img src={{ logo_url }} class="img-fluid"/></span>
```
[the remark this answers is not in the document]
Comment (jnation3406) [reply schedule/templates/downtime_list.html:16]: Yeah I guess that seems like a good idea

**Comment by jnation3406** [comment]
hmm the tests appear to fail if you don't have configdb set, even though it is mocked. I think it still needs a http:// in it or requests breaks even with the mocking. I'll add a test_settings file like we do with the obs portal to set some sensible values for testing...

**Comment by jnation3406** [comment]
Also, I think if you set DEBUG to True, then you don't need to run collectstatic first.

File: schedule/views.py [hunk 4]
```diff
@@ -1,25 +1,31 @@
...
-        return JsonResponse([dt.as_dict() for dt in f.qs.all()], safe=False)
+    def get_filterset_kwargs(self, filterset_class):
+        kwargs = super(DowntimeListView, self).get_filterset_kwargs(filterset_class)
+        # If there are no query parameters or the only query parameter is for pagination, default to
+        # filtering out downtimes in the past
+        if kwargs['data'] is None or (len(kwargs['data']) == 1 and 'page' in kwargs['data']):
+            kwargs['data'] = {
+                'ends_after': timezone.now()
+            }
+        return kwargs
```
[the remark this answers is not in the document]
Comment (jnation3406) [reply schedule/views.py:31]: weird codacy catches these things sometimes but misses them others.

**Comment by jnation3406** [comment]
I addressed all the comments. Couldn't find a way to test the admin pages without running collectstatic first, so I added that to the readme on how to run the tests.

**Review by eheinrich — APPROVED** [review]
Cool, looks good!

**Pull request merged**